ginivision/src/main/java/net/gini/android/vision/internal/pdf/RendererLollipop.java

```diff
@@ -139,6 +164,21 @@ public synchronized int getPageCount() {
         return pageCount;
     }
 
+    @Override
+    public boolean isPdfPasswordProtected() {
+        final PdfRenderer pdfRenderer;
+        try {
+            pdfRenderer = getPdfRenderer();
+        } catch (final SecurityException e) {
+            LOG.debug("PDF is password protected", e);
+            return true;
```

Comment:
Ok the exception is needed here, but I find it weird that a side effect of a method is used to determine the state of password protection.

Reply:
I know. Would be great if `PdfRenderer` would have a static method to check whether a pdf is protected or not. I will look into dealing with the `SecurityException` in a cleaner way. I think I can refactor it a bit.